interface/HHReweight5D.h

```diff
@@ -3,62 +3,91 @@
 ** author : L. Cadamuro (LLR)
 ** date   : 17 Feb 2017
 ** brief  : Compute a reweight coefficient to transform an input distribution into a general point in kl, kt plane. 31/05/2017: Updated to 5D reweight
+** Updated by D. Zuolo (MIB) and J. Motta (LLR) for Run 2 analisys - Apr 2021
+**      - the update is based on the approach explained in https://gitlab.cern.ch/hh/eft-benchmarks
+**      - the current version of this class allows to get the HHweight for specific EFT benchmarks, c2scans, or manually set couplings
+**      - the current version of this class allows to do the reweight both at LO and NLO
 */ 
 
-
 #ifndef HHREWEIGHT_H
 #define HHREWEIGHT_H
 
-#include <array>
+#include <cmath>
+#include <fstream>
 #include <iostream>
-#include <memory>
+#include <sstream>
 #include <string>
-#include <fstream>
 #include <utility>
+#include <vector>
 
-#include "TH1.h"
-#include "TH2.h"
 #include "TAxis.h"
 #include "TFile.h"
-
-#define NCOEFFSA 15
+#include "TH1.h"
+#include "TH2.h"
+#include "TH2D.h"
+#include "TMath.h"
 
 class HHReweight5D{
     
     public:
-        HHReweight5D(std::string coeffFile, const TH2* hInput, bool useAbsEta=true);
+        HHReweight5D(std::string coeffFile, const TH2* hInput, std::string EFTBMname, std::string year, std::string order, std::string uncertantie = "", bool cms_fake = false, bool useAbsEta=true);
+        HHReweight5D(std::string coeffFile, const TH2* hInput, std::string order, std::string uncertantie = "", bool useAbsEta=true);
```

Comment:
I put the same question below: I'm not sure I understand the deffirence between these two constructors.